distributed/scheduler.py

```diff
@@ -1441,14 +1448,26 @@ def add_worker(
             self.log_event("all", {"action": "add-worker", "worker": address})
             logger.info("Register %s", str(address))
 
-            yield comm.write(
-                {
-                    "status": "OK",
-                    "time": time(),
-                    "heartbeat-interval": heartbeat_interval(len(self.workers)),
-                    "worker-setups": self.worker_setups,
-                }
+            msg = {
+                "status": "OK",
+                "time": time(),
+                "heartbeat-interval": heartbeat_interval(len(self.workers)),
+                "worker-setups": self.worker_setups,
+            }
+
+            version_warning = version_module.error_message(
```

Comment:
Why generate the warning message in the scheduler? I'd send that information back to the client, and have the client generate the message itself (or perhaps put the values in the `VersionMismatchWarning` object and have the object format itself on warn).

Reply:
All of the information is centralized here.  It's not huge (even for large clusters) but I don't see a strong motivation to move it around.

Reply:
Ah, I guess that we only need to send the information if there's a problem though, so in common case this isn't a big deal.

Reply:
I'm going to pass on this for now in the interests of getting this in.  We can improve this in the future if necessary.